content/rationale.md

```diff
@@ -0,0 +1,85 @@
+title: Why complate?
+
+Why complate?
+=============
+
+## Abstraction
+
+When we are developing our frontend components, we can use complate markup
+abstractions ("macros") as an abstraction layer for our component. Consider
+the following macro:
+
+```jsx
+function Navbar (_props, ...children) {
+  return <nav>
+    {children}
+  </nav>
+}
+```
+
+We can then use our macro in our application and it will be replaced with the
+markup we have defined.
+
+```jsx
+<Navbar />
+```
+
+This is _extremely powerful_ and solves a major pain point when writing
+maintainable frontend code. Requirements change constantly. When a new request
+comes, or the corporate design has changed yet again, I can change the markup
+of my component _without breaking any part of my application that uses the
+component_.
+
+Another major benefit is that it is possible to write complex frontend
+components and others can _use_ the components easily, without having to
+understand how the component works under the hood.
+
+## Composition
+
+complate allows us to write components and then explicitly compose them
+together. For instance, when I write a navbar component, I can reuse the
+component in multiple projects and change the content of the navbar without
+affecting it.
+
+The composition in complate looks very similar to the composition that is
+inherent in HTML: child components are passed into the parent component
+by defining them within the parent component tag.
+
+```jsx
+function NavbarItem ({ href, current }, ...children) {
+	const className = current ? 'currentPage' : ''
```

Comment:
FYI: `null` instead `''` of would omit the `class` attribute entirely. (Also, CamelCase in class names is a crime. 🙂 )

Reply:
should I use `class={current && 'current'}`

Reply:
I just think that `const className = current ? 'current' : null` doesn't look so nice (I don't like explicitly using null). Is there a better way? Could you use `className = current && 'current'` ? Although I don't know if that would be more readable.

Reply:
Nothing wrong with explicit `null` IMO. 🤷‍♂

`&&` is probably idiomatic in the JSX/React world, but I agree that it might be confusing in this context (it's not necessarily obvious that JS's logical operators work this way - plus it might be brittle; think `0`).

Reply:
I think `className = current ? 'current' : null` is definitely easier to read.